src/inmanta/execute/runtime.py

```diff
@@ -1014,6 +1014,9 @@ def _unsafe_execute(self) -> None:
     def execute(self) -> None:
         try:
             self._unsafe_execute()
+        except NotFoundException as e:
+            # In case of a NotFoundException, keep all the info of the exception as is
+            raise e
         except RuntimeException as e:
             e.set_statement(self.owner)
```

Comment:
I wonder, it is ever appropriate to overwrite the statement if one is already set? 
-> If it is not appropriate, we can simplify all of this very significantly but not overwriting. 

Second, how do we know that all `NotFoundExceptions` that pass by here do have a statement set?